Allow apps to set a custom inspection name for web extension background and popup web views

kiaraarose opened the pull request:
#### e327d4468b6d6c0366404ed9e35dd215d4062610
<pre>
Allow apps to set a custom inspection name for web extension background and popup web views
<a href="https://bugs.webkit.org/show_bug.cgi?id=273161">https://bugs.webkit.org/show_bug.cgi?id=273161</a>
<a href="https://rdar.apple.com/125874004">rdar://125874004</a>

Reviewed by Timothy Hatcher.

Add API to allow apps to set a custom inspection name for background and popup web views.
Also, add a `nameOverride()` method to WebPageDebuggable. This will allow us to keep
track of overrides that are set for the web pages.

* Source/JavaScriptCore/inspector/remote/RemoteInspectionTarget.h:
* Source/JavaScriptCore/inspector/remote/RemoteInspectorConstants.h:
* Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm:
(Inspector::RemoteInspector::listingForInspectionTarget const):
Set the name override on the listing.

* Source/WebCore/en.lproj/Localizable.strings:
Add localized strings for text shown when inspecting the extension popup.

* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionAction.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionAction.mm:
(-[_WKWebExtensionAction inspectionName]):
(-[_WKWebExtensionAction setInspectionName:]):

* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.mm:
(-[_WKWebExtensionContext inspectionName]):
(-[_WKWebExtensionContext setInspectionName:]):

* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionControllerConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionDataRecord.h
Remove guards since _WKWebExtensionDataRecord.errors API has made it into the build.

* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm:
(WebKit::WebExtensionAction::popupWebViewInspectionName):
(WebKit::WebExtensionAction::setPopupWebViewInspectionName):
(WebKit::WebExtensionAction::popupWebView):
Set the _remoteInspectionNameOverride for the popup web view.

* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::loadBackgroundWebView):
(WebKit::WebExtensionContext::backgroundWebViewInspectionName):
(WebKit::WebExtensionContext::setBackgroundWebViewInspectionName):

* Source/WebKit/UIProcess/Extensions/WebExtensionAction.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
* Source/WebKit/UIProcess/Inspector/WebPageDebuggable.cpp:

(WebKit::WebPageDebuggable::name const):
* Source/WebKit/UIProcess/Inspector/WebPageDebuggable.h:

Canonical link: <a href="https://commits.webkit.org/277937@main">https://commits.webkit.org/277937@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/112b9d95b3d8984c4a87063613cbdf4d18bb154f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49002 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28215 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51969 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51690 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45072 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51307 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34184 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25743 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/40056 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/50047 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/25858 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/42249 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/21164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/23316 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/43419 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7057 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/42301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/45242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/43924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53600 "Built successfully") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/48493 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24053 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/20306 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/47370 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/25316 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/42458 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/46339 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26125 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/55988 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7011 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/25036 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/11521 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm:
```diff
@@ -646,6 +646,22 @@ - (void)_otherPopoverWillShow:(NSNotification *)notification
+
+void WebExtensionAction::setPopupWebViewInspectionName(const String& name)
+{
+    if (m_popupWebView)
```
No need to nil check since it is ObjC.

xeenon commented on Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
```diff
@@ -3212,6 +3209,27 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
+
+void WebExtensionContext::setBackgroundWebViewInspectionName(const String& name)
+{
+    if (m_backgroundWebView)
```
No need to nil check.

kiaraarose committed:
Allow apps to set a custom inspection name for web extension background and popup web views
https://bugs.webkit.org/show_bug.cgi?id=273161
rdar://125874004

Reviewed by Timothy Hatcher.

Add API to allow apps to set a custom inspection name for background and popup web views.
Also, add a `nameOverride()` method to WebPageDebuggable. This will allow us to keep
track of overrides that are set for the web pages.

* Source/JavaScriptCore/inspector/remote/RemoteInspectionTarget.h:
* Source/JavaScriptCore/inspector/remote/RemoteInspectorConstants.h:
* Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm:
(Inspector::RemoteInspector::listingForInspectionTarget const):
Set the name override on the listing.

* Source/WebCore/en.lproj/Localizable.strings:
Add localized strings for text shown when inspecting the extension popup.

* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionAction.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionAction.mm:
(-[_WKWebExtensionAction inspectionName]):
(-[_WKWebExtensionAction setInspectionName:]):

* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionContext.mm:
(-[_WKWebExtensionContext inspectionName]):
(-[_WKWebExtensionContext setInspectionName:]):

* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionControllerConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebExtensionDataRecord.h
Remove guards since _WKWebExtensionDataRecord.errors API has made it into the build.

* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm:
(WebKit::WebExtensionAction::popupWebViewInspectionName):
(WebKit::WebExtensionAction::setPopupWebViewInspectionName):
(WebKit::WebExtensionAction::popupWebView):
Set the _remoteInspectionNameOverride for the popup web view.

* Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm:
(WebKit::WebExtensionContext::loadBackgroundWebView):
(WebKit::WebExtensionContext::backgroundWebViewInspectionName):
(WebKit::WebExtensionContext::setBackgroundWebViewInspectionName):

* Source/WebKit/UIProcess/Extensions/WebExtensionAction.h:
* Source/WebKit/UIProcess/Extensions/WebExtensionContext.h:
* Source/WebKit/UIProcess/Inspector/WebPageDebuggable.cpp:

(WebKit::WebPageDebuggable::name const):
* Source/WebKit/UIProcess/Inspector/WebPageDebuggable.h:

Canonical link: https://commits.webkit.org/277937@main

dcrousso commented on Source/WebCore/page/PageDebuggable.cpp:
```diff
@@ -46,9 +46,6 @@ PageDebuggable::PageDebuggable(Page& page)
 
 String PageDebuggable::name() const
 {
-    if (!m_nameOverride.isNull())
-        return m_nameOverride;
```
why was this not enough?  is there a situation where we need _both_ the `name` and `nameOverride`?

dcrousso commented on Source/WebKit/UIProcess/Inspector/WebPageDebuggable.cpp:
```diff
@@ -43,9 +43,6 @@ WebPageDebuggable::WebPageDebuggable(WebPageProxy& page)
 
 String WebPageDebuggable::name() const
 {
-    if (!m_nameOverride.isNull())
-        return m_nameOverride;
```
ditto
